rpc/rpc-lib/src/rpc-clnt.c

```diff
@@ -1753,7 +1759,9 @@ rpc_clnt_submit(struct rpc_clnt *rpc, rpc_clnt_prog_t *prog, int procnum,
         if (rpcreq) {
```

Comment:
The only case where `rpcreq` can be NULL is if `rpc`, `proc` or `frame` are NULL. I think that shouldn't happen (in which case the test should be removed), but if we want to keep it, even in this case we should call the `cbkfn` with a fake static rpcreq. Otherwise the possibility of returning -1 and not having destroyed the `frame` is still there.

Reply:
Ahh, yes missed that. Will update it.

Reply:
You can't simply remove the check. If we are sure that `rpc`, `proc` and `frame` won't be NULL (i.e. all callers pass non NULL values), we also need to remove the check at line 1623. Otherwise we need to also initialize `rpcreq` in that case.

Reply:
Yeah, I added that part in my recent commit, i.e. initialized `rpcreq` in the case if either of `rpc`, `proc`, `frame` is NULL.

Reply:
Only one last suggestion: given that we do the `rpcreq_static` initialization twice, it would be better to avoid that duplication of code and move it here. If `rpcreq` is NULL here, we initialize `rpcreq_static`. Otherwise we use the existing one.

Reply:
Makes sense, done that!